Reference/Searching/Examine/Quick-Start/index.md

```diff
@@ -25,39 +23,78 @@ We will make it possible to 'search' on the _People_ page, by adding a search ba
-### Retrieving the Search Term
+### Using route hijacking
 
-Right below the form, add the following:
+Now that we can post to the people view, we want a bit more control, we probably want a custom ViewModel, with all the search results, when people try to search for something!
+We will use route hijacking by creating a render controller, for more information about route hijacking and render controllers, you can read our documentation: https://our.umbraco.com/documentation/reference/routing/custom-controllers/
+
+Lets start by creating a PeopleController, and letting it derive from `RenderController`, and then adding an Index method.
+:::note Its important to name our Controller by the convention `NameOfViewController` so in our case our view is named People, so we call it PeopleController
 ```csharp
-<div>
-    @{
-        var searchTerm = string.Empty;
-        searchTerm = string.IsNullOrEmpty(Context.Request.Query["query"])
-            ? string.Empty
-            : Context.Request.Query["query"];
-        if (searchTerm == string.Empty)
-        {
-            <p>Enter search term</p>
-        }
-        else
-        {
-            //perform the search
-        }
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Umbraco.Cms.Core.Web;
+using Umbraco.Cms.Web.Common.Controllers;
+
+
+public class PeopleController : RenderController
+{
+    public PeopleController(
+    ILogger<RenderController> logger,
+    ICompositeViewEngine compositeViewEngine,
+    IUmbracoContextAccessor umbracoContextAccessor)
+    : base(
+        logger,
+        compositeViewEngine,
+        umbracoContextAccessor)
+    {
     }
-</div>
+
+    public override IActionResult Index()
+    {
+        return CurrentTemplate(CurrentPage);
+    }
+}
+```
+
+Now we have created our controller, and your code will now do the exact same thing as before you created the Controller!
+Lets change that and start adding logic to the index method.
+
+### Adding a Service that handles our search logic
+To search anything from our controller, we first need to create a service we can use, that handles the actual search logic, lets create an interface first.
+
+```csharp
+using Umbraco.Cms.Core.Models.PublishedContent;
+
+public interface ISearchService
+{
+    IEnumerable<IPublishedContent> SearchContentNames(string query);
+}
+
+```
+
+So this method can take in a query string, and then return content that matches!
+Lets create our service that uses this interface
+```csharp
+using Umbraco.Cms.Core.Models.PublishedContent;
+using Umbraco.Cms.Web.UI.Services;
+
+public class SearchService : ISearchService
+{
+    public IEnumerable<IPublishedContent> SearchContentNames(string query) => throw new NotImplementedException();
+}
 ```
-Here we are getting the request query from the form. If the string is empty we ask them to submit a term, otherwise we will perform the search.
+:::note Remember to register this service in your Startup.cs file!
```

Comment:
```suggestion

And finally register the service in `Startup`.

```csharp
public void ConfigureServices(IServiceCollection services)
{
    // ... (removed for abbreviation)
    services.AddTransient<ISearchService, SearchService>();
}
```

```